File: src/sensor_nivel.py
class SensorNivel:
    def __init__(
        self,
        tag: str,
        valor: float,
        unidade: str = ""
    ):
        self.tag = tag
        self.valor = valor
        self.unidade = unidade

    def resumo(self) -> str:
        resultado = f"{self.tag}: {self.valor:g}"

        if self.unidade:
            resultado += f" {self.unidade}"

        return resultado

File: src/sensor_nivel.cpp
#include "sensor_nivel.hpp"

#include <sstream>

namespace {
std::string formatarValor(double valor) {
    std::ostringstream saida;
    saida << valor;
    return saida.str();
}
}  // namespace

SensorNivel::SensorNivel(std::string tagInicial, double valorInicial)
    : tag(tagInicial), valor(valorInicial) {
}

std::string SensorNivel::resumo() const {
    return tag + ": " + formatarValor(valor);
}
